worker/src/router.rs

```diff
@@ -190,7 +190,11 @@ impl<'a, D: 'a> Router<'a, D> {
 
     /// Register an HTTP handler that will exclusively respond to HEAD requests. Enables the use of
     /// `async/await` syntax in the callback.
-    pub fn head_async<T>(mut self, pattern: &str, func: fn(Request, RouteContext<D>) -> T) -> Self
+    pub fn head_async<T>(
+        mut self,
+        pattern: &str,
+        func: impl Fn(Request, RouteContext<D>) -> T + 'static,
+    ) -> Self
     where
```

Comment:
Alternatively, this doesn't need to be an APIT, but could also just be a generic parameter.

```suggestion
    pub fn head_async<T>(
        mut self,
        pattern: &str,
        func: F,
    ) -> Self
    where
        F: Fn(Request, RouteContext<D>) -> T + 'static,
```